### ISSUE TITLE
bug: undefined stripped from query result

### ISSUE THREAD

**Issue by ibrahimyaacob92** [trpc/trpc#4390]
### Provide environment information

project running on turborepo

packages/api/package.json
```
  "dependencies": {
    "db": "*",
    "@trpc/client": "^10.20.0",
    "@trpc/server": "^10.20.0",
    "superjson": "1.9.1",
    "zod": "^3.21.4"
  },
```

app/plasmo/package.json
```
  "dependencies": {
    "@clerk/chrome-extension": "^0.2.7",
    "@plasmohq/messaging": "^0.2.0",
    "@tanstack/react-query": "^4.29.5",
    "@trpc/client": "^10.21.1",
    "@trpc/react": "^9.27.4",
    "@trpc/server": "^10.21.1",
    "plasmo": "0.69.1",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-router-dom": "^6.10.0"
  },
  "devDependencies": {
    "@plasmohq/prettier-plugin-sort-imports": "3.6.4",
    "@types/chrome": "0.0.233",
    "@types/node": "18.15.12",
    "@types/react": "18.0.37",
    "@types/react-dom": "18.0.11",
    "api": "*",
    "db": "*",
    "prettier": "2.8.7",
    "typescript": "5.0.4"
  },
```

### Describe the bug

I encounter this issue when working with TRPC React under a Turborepo. But I works fine on Next

I see that its been merged but still appearing on my end
https://github.com/trpc/trpc/issues/4049

![image](https://github.com/trpc/trpc/assets/58116068/5dae13ce-99d4-48f5-bb35-ad232f13c825)
![image](https://github.com/trpc/trpc/assets/58116068/8e191fcd-8d19-4294-9846-9c71661a3d81)

discussed here: https://discord.com/channels/867764511159091230/1102212861259943986

### Link to reproduction

https://github.com/ibrahimyaacob92/t3-plasmo-turbo

### To reproduce

Install the dependencies, check out apps\plasmo\src\popup\components\Todos.tsx the type of listOfTodos is `Todos[]` without `undefined`. 

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

**Comment by kamilogorek** [comment]
Are you sure that your lockfile is not forcing you to use `10.20.0` mentioned in your issue? The fix has been released in `10.21.0`.

**Comment by KATT** [comment]
- Duplicate of #4049
- Fixed in #4187 / [10.21.0](https://github.com/trpc/trpc/releases/tag/v10.21.0)


Closing. Feel free to re-open if it doesn't work after updating your deps.
